app/src/main/java/com/github/se/orator/ui/profile/EditProfile.kt

```diff
@@ -1,2 +1,268 @@
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.layout.statusBarsPadding
+import androidx.compose.material.AlertDialog
+import androidx.compose.material.BottomNavigation
+import androidx.compose.material.BottomNavigationItem
+import androidx.compose.material.Button
+import androidx.compose.material.IconButton
+import androidx.compose.material.OutlinedTextField
+import androidx.compose.material.Scaffold
+import androidx.compose.material.Text
+import androidx.compose.material.TopAppBar
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.unit.dp
+import com.github.se.orator.R
+import com.github.se.orator.model.profile.UserProfileViewModel
+import com.github.se.orator.ui.navigation.NavigationActions
+
+@Composable
+fun EditProfileScreen(
+    navigationActions: NavigationActions,
+    userProfileViewModel: UserProfileViewModel
+) {
+    // Fetch the user's profile data
+    val userProfile by userProfileViewModel.userProfile.collectAsState()
+
+    // States for username, bio, and dialog visibility
+    var isDialogOpen by remember { mutableStateOf(false) }
+    var updatedUsername by remember { mutableStateOf(userProfile?.name ?: "") }
+    var updatedBio by remember { mutableStateOf(userProfile?.bio ?: "") }
+
+    // Intent launcher to capture photo or pick image from gallery
+    val context = LocalContext.current
+    val takePictureLauncher = rememberLauncherForActivityResult(
+        contract = ActivityResultContracts.TakePicturePreview()
+    ) { bitmap ->
+        // Handle the profile picture update here
+        bitmap?.let {
+            // Assuming you have a function in the ViewModel to upload the profile picture
+            userProfileViewModel.uploadProfilePicture(userProfile?.uid ?: "", Uri.EMPTY)  // Replace with actual URI logic
+        }
+    }
+
+    val pickImageLauncher = rememberLauncherForActivityResult(
+        contract = ActivityResultContracts.GetContent()
+    ) { uri ->
+        uri?.let { userProfileViewModel.uploadProfilePicture(userProfile?.uid ?: "", it) }
+    }
+
+    Scaffold(
+        topBar = {
+            TopAppBar(
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .statusBarsPadding(),
+                backgroundColor = Color.White,
```

Comment:
The color should adapt to the theme (light / dark) : i don't think a specific color should be specified !